armory/datasets/config_load.py

```diff
@@ -49,10 +48,16 @@ def actual_loader(
 
     if preprocessor_name is None:
         preprocessor = None
+        if name in preprocessing.list_registered():
+            preprocessor = preprocessing.get(name)
     else:
         preprocessor = preprocessing.get(preprocessor_name)
 
-    return generator.ArmoryDataGenerator(
+    if preprocessor_kwargs is None:
+        preprocessor_kwargs = {}
+    preprocessing_fn = lambda x: preprocessor(x, **preprocessor_kwargs)
```

Comment:
if there aren't any preprocessor_kwargs, we should probably just set `preprocessing_fn = preprocessor`. It will make debugging a lot easier if there are errors in the preprocessor, I think.

Reply:
agreed, added a commit to address